apps/experiments/models.py

```diff
@@ -639,8 +640,32 @@ def send_bot_message(self, instruction_prompt: str, fail_silently=True):
         bot_message = bot_prompt_for_user(self, prompt_instruction=instruction_prompt)
         try_send_message(self, message=bot_message, fail_silently=fail_silently)
 
+    def get_participant_scheduled_messages(self):
+        from apps.events.models import ScheduledMessage
+
+        messages = ScheduledMessage.objects.filter(participant=self.participant, team=self.team)
+        scheduled_messages_str = ""
+        for message in messages:
+            schedule_info = message.action.params
+            name = schedule_info["name"]  # TODO: We need to make sure there aren't duplicate schedule names
+            frequency = schedule_info["frequency"]
+            time_period = schedule_info["time_period"]
+            repetitions = schedule_info["repetitions"]
+            schedule = f"{name}: Every {frequency} {time_period}, {repetitions} times"
+            if time_period not in ["hour", "day"]:
+                weekday = message.next_trigger_date.strftime("%A")
+                schedule = f"{name}: Every {frequency} {time_period} on {weekday} for {repetitions} times"
+            next_trigger = pretty_date(message.next_trigger_date)
+            schedule = f"{schedule} (next trigger is {next_trigger})"
+            scheduled_messages_str = schedule if not scheduled_messages_str else f"{scheduled_messages_str},{schedule}"
+        return scheduled_messages_str
+
     def get_participant_data(self):
-        return self.experiment.get_participant_data(self.participant)
+        scheduled_messages_str = self.get_participant_scheduled_messages()
+        participant_data = self.experiment.get_participant_data(self.participant)
+        return f"{participant_data}. Scheduled messages:\n{scheduled_messages_str}"
 
     def get_participant_data_json(self):
-        return json.dumps(self.experiment.get_participant_data(self.participant), indent=2)
+        data = self.experiment.get_participant_data(self.participant) or {}
+        data["schedules"] = self.get_participant_scheduled_messages()
```

Comment:
shouldn't this be a more JSONy format instead of the string format?

Reply:
yeah I realized there was some work missing in this area. I made a small update here: https://github.com/dimagi/open-chat-studio/pull/416/commits/eef15a38223fb73a2ebe9a06783025bd06d34f49

Reply:
I think a JSON object would be better because otherwise if anyone wants to do anything with the schedule information they have to parse it out of the strings.

Reply:
Cool cool

Reply:
https://github.com/dimagi/open-chat-studio/pull/416/commits/3c3b08c47bde95a61768738d1336833b3bd65c43